Update Select Cypher versions guide with relevant links to Operations Manual and Upgrade and migration guide (#1327)
neo4j/docs-cypher: modules/ROOT/pages/queries/select-version.adoc

This setting determines the default language for new databases where it has not been specified as part of a `CREATE` or `ALTER` database command.
For example, `config_db.query.default_language=CYPHER_25` will set Cypher 25 as the default language for a DBMS.
For more information about using configuration settings, see the link:{neo4j-docs-base-uri}/operations-manual/current/configuration/[Operations Manual -> Configuration].

The table below outlines which Cypher version is assigned to databases in different upgrade or installation scenarios:

[cols="3", options="header"]
|===
| Scenario | `db.query.default_language` | Databases

| *Standard DBMS upgrade to Neo4j 2025.06+*
| Unset
| *Existing system database:* `CYPHER 5` +
*Existing user databases:* `CYPHER 5` +
*New user databases default:* `CYPHER 5`

| *Custom DBMS upgrade to Neo4j 2025.06+*
| Manually set to `CYPHER_25` by administrator
| *Existing system database:* `CYPHER 5` +
*Existing user databases:* `CYPHER 5`  +
*New user databases default:* `CYPHER 25`

| *New installation of Neo4j 2025.06+*
| Set to `CYPHER_5`
| *New system database:* `CYPHER 5` +
*New user databases default:* `CYPHER 5`

> |===

Reviewer: I like this kind of table, if we don't want to keep it here maybe it should be added to the `cypher-version-configuration` page?

Author: @renetapopova added a similar one to the upgrade and migration guide, to which this PR links.

https://github.com/neo-technology/neo4j-upgrade-migration-guide/commit/d1c219d46513b1ce2341cb3427f167fccd9e620b

Reviewer: As per https://github.com/neo4j/docs-operations/pull/2372/files/589c50edd20c53367d35f64e2fb14446c450946f#r2171431101, it should be on the Upgrade and migration guide

Reviewer: not sure I understand that ones current table though 
![image](https://github.com/user-attachments/assets/f2592f44-7309-470a-bf02-35ddf4258d5f)

the table here were more explicit in what the different scenarios were which was nice, so maybe we need to update that table somewhat.

Author: cc @renetapopova

> * link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/alter-databases/#alter-database-default-language[Alter the default Cypher version of an existing database]

Reviewer: do we also want to have a link to `== Alter the default Cypher version of a composite database` in [`alter-composite-databases.adoc`](https://github.com/neo4j/docs-operations/pull/2372/files#diff-040de165384edb8f0806b47ff43e1ece31c6ed96baf6150cdca3064d5ff30a87) (though that would need a tag added in that PR so it could be referenced)

Author: Have added (technically you don't need an anhor/tag in order to reference. If you dont have one, you just have to prepend "_" to whatever the header title is).

> * link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Show the default Cypher version of a database]

Reviewer: I guess we're skipping linking to the show alias section as it doesn't have a specific example/section for default language?

Author: There is a show example here: https://neo4j-docs-operations-2372.surge.sh/operations-manual/2025.06/database-administration/aliases/manage-aliases-standard-databases/#set-default-language-for-remote-database-aliases

But that is already linked.
If there is a separate show default language for alias section, we can add it

Reviewer: > If there is a separate show default language for alias section, we can add it

there isn't one, it's just one more column in some of the other examples in the show section 🤷

> * link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#_grant_privilege_to_modify_the_default_language_of_standard_databases[Grant privilege to modify the default language of standard databases]

Reviewer: do we also want to link to the alter composite database privilege for the privilege to modify default language of composite databases?
`=== Grant privilege to modify composite databases` in [`dbms-administration.adoc`](https://github.com/neo4j/docs-operations/pull/2372/files#diff-00d1dc34b4a58b5725ccd99993f147c046fe28ced1c604e4ed1989ecc7c8aa50), will also need a tag to be able to be referenced

* link:{neo4j-docs-base-uri}/operations-manual/current/configuration/cypher-version-configuration/[Configure the Cypher default version]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/create-databases/#_set_a_default_cypher_version_for_a_standard_database[Set a default Cypher version for a standard database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/create-composite-databases/#composite-databases-default-language[Set a default Cypher version for a composite database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#set-default-language-for-remote-database-aliases[Set a default Cypher version for remote database aliases]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/alter-databases/#alter-database-default-language[Alter the default Cypher version of an existing database]

> * link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/alter-composite-databases/#_alter_the_default_cypher_version_of_a_composite_database[Alter the default Cypher version of a composite database]

Reviewer: it got it's own tag now
```suggestion
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/alter-composite-databases/#alter-default-language-composite-database[Alter the default Cypher version of a composite database]
```

This is true for link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/#manage-database-systems[system, standard, and composite] Neo4j databases.
However, it is possible to set a different default language on both new and existing system, standard, and composite databases.

> For more information about how to change and view the default language of new and existing databases and remote database aliases, and for how to set the necessary privilege to change the default language of a standard or composite database, see the following links in the Operations Manual:

Reviewer: feels slightly odd to only add privileges for the database cases and not the alias one 🤔 the alias privilege page doesn't talk about default language specifically but `=== Grant privilege to modify aliases` will be the relevant privilege for altering default language on a remote alias 🤷

Author: Ok, then please say exactly which privileges we should add to this list.

Reviewer: Do we need to mention privileges at all? We could just leave it out 🤷 

But if we want to cover all the needed privileges:
* For creation of a standard or composite database, or a remote alias with a default language you only need the respective create privilege (CREATE DATABASE, CREATE COMPOSITE DATABASE, CREATE ALIAS).
* For altering a standard database you need the SET DATABASE DEFAULT LANGUAGE privilege (as you have in the list below already)
* For altering a composite database you need the ALTER COMPOSITE DATABASE privilege (which we already added below)
* For altering a remote alias you need the ALTER ALIAS privilege, this is the same one as altering any other part of the alias so no different than before (just one more capability)

Reviewer: as discussed on slack, lets just have the new privileges in the list and update the sentence here again

> For more information about how to change and view the default language of new and existing databases and remote database aliases, and for how to set the necessary privilege to change the default language of a standard or composite database, see the following links in the Operations Manual:

Reviewer: could be simplified to 
```suggestion
For more information about how to change and view the default language of new and existing databases and remote database aliases, and for how to set the necessary privileges to change the default language of a database, see the following links in the Operations Manual:
```

* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/alter-databases/#alter-database-default-language[Alter the default Cypher version of an existing database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/alter-composite-databases/#_alter_the_default_cypher_version_of_a_composite_database[Alter the default Cypher version of a composite database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#alter-default-language-remote-database-alias[Alter the default Cypher version of a remote database alias]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Show the default Cypher version of a database]
* link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#_grant_privilege_to_modify_the_default_language_of_standard_databases[Grant privilege to modify the default language of standard databases]

> * * link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#_grant_privilege_to_modify_composite_databases[Grant privilege to modify composite databases] (required in order to alter the default language of a composite database)

Reviewer: why the double `*`?

Author: Because mistake.

This is true for link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/#manage-database-systems[system, standard, and composite] Neo4j databases.
However, it is possible to set a different default language on both new and existing system, standard, and composite databases.

To select a default Cypher version when creating a database, add `DEFAULT LANGUAGE <language version>` to the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/create-databases/[`CREATE DATABASE`] statement.

.Select default Cypher version when creating a database
[.tabbed-example]
=====
[.include-with-Cypher-25]
======

.Cypher 25
[source,cypher]
----

> CREATE DATABASE actors DEFAULT LANGUAGE CYPHER 25

Reviewer: ```suggestion
CREATE DATABASE actors SET DEFAULT LANGUAGE CYPHER 25
```
I prefer the explicit SET when available (as per https://github.com/neo4j/docs-cypher/pull/1123#discussion_r2168517271)

* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/alter-databases/#alter-database-default-language[Alter the default Cypher version of an existing database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/alter-composite-databases/#alter-default-language-composite-database[Alter the default Cypher version of a composite database]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#alter-default-language-remote-database-alias[Alter the default Cypher version of a remote database alias]
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Show the default Cypher version of a database]
* link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#_grant_privilege_to_modify_the_default_language_of_standard_databases[Grant privilege to modify the default language of standard databases]

> * link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#_grant_privilege_to_modify_composite_databases[Grant privilege to modify composite databases] (required in order to alter the default language of a composite database)

Reviewer: ```suggestion
* link:{neo4j-docs-base-uri}/operations-manual/authentication-authorization/dbms-administration/#grant-privilege-alter-composite-database[Grant privilege to modify composite databases] (required in order to alter the default language of a composite database)
```
